### PR TITLE
[Thesis] Support vm dependencies in Winery

### PR DESCRIPTION
**Pull request by barlasemre**
Start and end date: 01.04.2020 - 26.11.2020 
Contributor: Emre Barlas, @barlasemre 
Supervisor: Lukas Harzenetter, @lharzenetter

This pull request contains the changes to integrate the [Packager Framework](https://github.com/zimmerml/TOSCA_packager) functionality in Winery via Export CSAR process. 

In order to support the dependency resolving of the VM Images, we have to download and include the relevant ISO files when creating a CSAR export. If the files are already existing in our repository, find them based on their naming pattern and use it for the export. Same applies for the Docker images. Docker containers will be builded during the process of exporting.

- [ ] Ensure that you followed http://eclipse.github.io/winery/dev/ToolChain#github---prepare-pull-request. Especially, we require **a single commit**
- [ ] Ensure that the commit message is [a good commit message](https://github.com/joelparkerhenderson/git_commit_message)
- [ ] Ensure to use auto format in **all** files
- [ ] Ensure that you appear in `NOTICE` at Copyright Holders
- [ ] Tests created for changes
- [ ] Documentation updated (if needed)
- [ ] Screenshots added (for UI changes)

### CODE REVIEW

**Comment by genie-winery** [comment]
Can one of the admins verify this patch?

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/BackendUtils.java [hunk 1]
```diff
@@ -1054,6 +1054,25 @@ public static void copyIdToFields(HasIdInIdOrNameField element, DefinitionsChild
         }
     }
 
+    public static void fixReferences(Definitions element, DefinitionsChildId id) {
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/BackendUtils.java:1057]: What are you doing here? There is already a method which does something similar: `BackendUtils::synchronizeReferences`

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/CsarExporter.java [hunk 2]
```diff
@@ -214,6 +247,77 @@ public String writeCsar(IRepository repository, DefinitionsChildId entryId, Outp
...
+    // TODO: Need to be 
+    private void updateTopologyTemplate(DefinitionsChildId entryId, IRepository repository) throws IOException {
+        TServiceTemplate serviceTemplate = repository.getElement(entryId);
+
+        @NonNull List<TNodeTemplate> nodeTemplates = serviceTemplate.getTopologyTemplate().getNodeTemplates();
+
+        TTopologyTemplate topologyTemplateCopy = serviceTemplate.getTopologyTemplate();
+
+        nodeTemplates.forEach(tNodeTemplate -> {
+            boolean ubuntuFlag = false;
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/CsarExporter.java:259]: This is very hard coded, can you think of a better architecture using plugins? For example, that you register available plugins that decide which elements and how to transform them?

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/CsarExporter.java [hunk 3]
```diff
@@ -214,6 +247,77 @@ public String writeCsar(IRepository repository, DefinitionsChildId entryId, Outp
...
+            boolean dockerFlag = false;
+            if (tNodeTemplate.getName().startsWith("Ubuntu-VM")) {
+                ubuntuFlag = true;
+            }
+
+            if (tNodeTemplate.getDeploymentArtifacts() != null) {
+                @NonNull List<TDeploymentArtifact> deploymentArtifacts = tNodeTemplate.getDeploymentArtifacts().getDeploymentArtifact();
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/CsarExporter.java:266]: You can remove the annotation here.

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/CsarExporter.java [hunk 4]
```diff
@@ -214,6 +247,77 @@ public String writeCsar(IRepository repository, DefinitionsChildId entryId, Outp
...
+
+                    if (deploymentArtifact.getArtifactType().getLocalPart().equals("DockerContainerArtifact")) {
+                        dockerFlag = true;
+                    }
+                }
+            }
+
+            if (dockerFlag || ubuntuFlag) {
+                TNodeTemplate duplicatedNode = BackendUtils.clone(tNodeTemplate);
+                duplicatedNode.setId(tNodeTemplate.getName());
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/CsarExporter.java:279]: This is already set by the clone method...

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/CsarExporter.java [hunk 5]
```diff
@@ -214,6 +247,77 @@ public String writeCsar(IRepository repository, DefinitionsChildId entryId, Outp
...
+                duplicatedNode.setName(tNodeTemplate.getName());
+
+                QName tNodeQtype = tNodeTemplate.getTypeAsQName();
+                QName qName = new QName(tNodeQtype.getNamespaceURI(), tNodeQtype.getLocalPart() + "-self", tNodeQtype.getPrefix());
+
+                duplicatedNode.setType(qName);
+
+                if (dockerFlag) {
+                    @Nullable QName artifactRef = duplicatedNode.getDeploymentArtifacts().getDeploymentArtifact().get(0).getArtifactRef();
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/CsarExporter.java:288]: What if there are multiple DAs attached?

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/CsarExporter.java [hunk 6]
```diff
@@ -214,6 +247,77 @@ public String writeCsar(IRepository repository, DefinitionsChildId entryId, Outp
...
+                    artifactTemplateList.add(new TDeploymentArtifact.Builder(artifactTemplate.getName(), artifactTemplate.getType())
+                        .setArtifactRef(QName.valueOf("{http://opentosca.org/artifacttemplates}" + artifactTemplate.getId())).build());
+
+                    TDeploymentArtifacts DAs = new TDeploymentArtifacts.Builder(artifactTemplateList).build();
+
+                    duplicatedNode.setDeploymentArtifacts(DAs);
+                }
+
+                topologyTemplateCopy.getNodeTemplateOrRelationshipTemplate().add(duplicatedNode);
+                topologyTemplateCopy.getNodeTemplateOrRelationshipTemplate().remove(tNodeTemplate);
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/CsarExporter.java:305]: I would change the order here, first remove it, then add it again because it may happen that it deletes the element based on the `equals` method and I think that they are equal if the IDs are equal, but not quite sure. Just to be save.

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/CsarExporter.java [hunk 7]
```diff
@@ -214,6 +247,77 @@ public String writeCsar(IRepository repository, DefinitionsChildId entryId, Outp
...
+                topologyTemplateCopy.setNodeTemplates(topologyTemplateCopy.getNodeTemplates());
+                List<TRelationshipTemplate> oldOutgoingRelationships = ModelUtilities.getOutgoingRelationshipTemplates(topologyTemplateCopy, tNodeTemplate);
+
+                List<TRelationshipTemplate> newOutgoingRelationships
+                    = resetOutgoingRelationships(oldOutgoingRelationships, duplicatedNode);
+                topologyTemplateCopy.getNodeTemplateOrRelationshipTemplate().removeAll(oldOutgoingRelationships);
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/CsarExporter.java:311]: I guess that if you just update the source/target of the relationships, its enough, you don't have to create new ones.

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/SelfContainmentUtil.java [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+
+public class SelfContainmentUtil {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SelfContainmentUtil.class);
+
+    public Collection<DefinitionsChildId> manageSelfContainedDefinitions(Collection<DefinitionsChildId> referencedDefinitionsChildIds, IRepository repository) throws IOException, JAXBException, RepositoryCorruptException {
+        Collection<DefinitionsChildId> processedIds = new HashSet<>();
+
+        for (DefinitionsChildId childid : referencedDefinitionsChildIds) {
+            // FIXME: Resolving Ubuntu is based on its name in several places. Any better way to do it?
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/SelfContainmentUtil.java:61]: Node Type --> `OpenToscaBaseTypes` class: Add it without the version

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/SelfContainmentUtil.java [hunk 9]
```diff
@@ -0,0 +1,274 @@
...
+                        }
+                    }
+                    processedIds.add(selfContainedNodeType);
+                    continue;
+                }
+            }
+
+            if (childid instanceof ArtifactTemplateId) {
+                TArtifactTemplate element = repository.getElement(childid);
+                if (element.getType() != null && "DockerContainerArtifact".equals(element.getType().getLocalPart())) {
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/SelfContainmentUtil.java:82]: Same --> `OpenToscaBaseTypes`

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/SelfContainmentUtil.java [hunk 10]
```diff
@@ -0,0 +1,274 @@
...
+        Process process;
+        final ProcessBuilder pb = new ProcessBuilder(command);
+        pb.directory(new File(directoryPath));
+
+        pb.redirectOutput(ProcessBuilder.Redirect.INHERIT);
+        pb.redirectError(ProcessBuilder.Redirect.INHERIT);
+        pb.redirectInput(ProcessBuilder.Redirect.INHERIT);
+
+        process = pb.start();
+        if (0 == process.waitFor()) {
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/SelfContainmentUtil.java:146]: What are you doing here?

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/SelfContainmentUtil.java [hunk 11]
```diff
@@ -0,0 +1,274 @@
...
+            throw new RuntimeException();
+        }
+
+        return artifactTemplateId;
+    }
+
+    private void downloadIsoFile(String path, String version) throws IOException {
+        Files.createDirectories(Paths.get(path + "/files"));
+        path = path + "/files/dependency.iso";
+        URL website = new URL("http://old-releases.ubuntu.com//releases/" + version + ".0/ubuntu-" + version + "-server-amd64.iso");
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/SelfContainmentUtil.java:242]: Is this always true? Looks very error-prone

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/ToscaExportUtil.java [hunk 12]
```diff
@@ -193,6 +194,10 @@ private void checkConfig(ExportProperties propKey, Boolean bo) {
 
         Collection<DefinitionsChildId> referencedDefinitionsChildIds = repository.getReferencedDefinitionsChildIds(tcId);
 
+//        if (includeDependencies) {
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/ToscaExportUtil.java:197]: If yo don't need this here, remove it

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/ToscaExportUtil.java [hunk 13]
```diff
@@ -400,4 +405,8 @@ protected void addVisualAppearanceToCSAR(IRepository repository, TopologyGraphEl
             }
         }
     }
+
+    public enum ExportProperties {
+        REPOSITORY_URI
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/ToscaExportUtil.java:410]: Why do you need this here?

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/splitting/Splitting.java [hunk 14]
```diff
@@ -67,6 +67,51 @@
     private Map<TNodeTemplate, Boolean> visitedNodeTemplates = new HashMap<>();
     private Map<TNodeTemplate, Set<TNodeTemplate>> transitiveAndDirectSuccessors = new HashMap<>();
 
+    public static List<TRelationshipTemplate> resetOutgoingRelationships(List<TRelationshipTemplate> outgoingRel, TNodeTemplate newSource) {
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/splitting/Splitting.java:70]: Why do you need this here in the Splitting class?

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/splitting/Splitting.java [hunk 15]
```diff
@@ -275,8 +320,6 @@ public ServiceTemplateId composeServiceTemplates(String composedSolutionServiceT
 
     /**
      *
-     * @param serviceTemplateId
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/splitting/Splitting.java:278]: if you delete this here, remove the whole comment.

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/splitting/Splitting.java [hunk 16]
```diff
@@ -1068,35 +1100,6 @@ private TRelationshipType getMatchingRelationshipType(TRequirement requirement,
...
     }
 
-    /**
-     * Switch the source of a relationship to a new node template
-     *
-     * @param outgoingRel all relationships which has to be switched to a new source element
-     * @param newSource   the new source element
-     * @return list of the reassigned relationships
-     */
-    private List<TRelationshipTemplate> reassignOutgoingRelationships(List<TRelationshipTemplate> outgoingRel, TNodeTemplate newSource) {
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/splitting/Splitting.java:1078]: why did you move the code?

**Review by lharzenetter — CHANGES_REQUESTED** [review]

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/BackendUtils.java (hunk 1 above)
Comment (lharzenetter) [reply org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/BackendUtils.java:1034]: Please check this here again

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/SelfContainmentUtil.java (hunk 8 above)
Comment (lharzenetter) [reply org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/SelfContainmentUtil.java:141]: todo

File: .idea/codeStyles/Project.xml [hunk 17]
```diff
@@ -28,18 +28,9 @@
...
-      <option name="PACKAGES_IMPORT_LAYOUT">
-        <value>
-          <package name="" alias="false" withSubpackages="true" />
-          <package name="java" alias="false" withSubpackages="true" />
-          <package name="javax" alias="false" withSubpackages="true" />
-          <package name="kotlin" alias="false" withSubpackages="true" />
-          <package name="" alias="true" withSubpackages="true" />
+          <package name="java.util" withSubpackages="false" static="false" />
+          <package name="kotlinx.android.synthetic" withSubpackages="true" static="false" />
+          <package name="io.ktor" withSubpackages="true" static="false" />
```
Comment (lharzenetter) [inline .idea/codeStyles/Project.xml:33]: Please revert the changes here

File: org.eclipse.winery.model.tosca/src/main/java/org/eclipse/winery/model/tosca/TTopologyTemplate.java [hunk 18]
```diff
@@ -114,7 +114,16 @@ public void setNodeTemplates(List<TNodeTemplate> nodeTemplates) {
...
         Objects.requireNonNull(id);
         return this.getNodeTemplates().stream()
-            .filter(x -> id.equals(x.getId()))
+            .filter(x -> id.contains(x.getId()) || x.getId().contains(id))
+            .findAny()
+            .orElse(null);
+    }
+
+    @Nullable
+    public TNodeTemplate getNodeTemplateByName(String name) {
```
Comment (lharzenetter) [inline org.eclipse.winery.model.tosca/src/main/java/org/eclipse/winery/model/tosca/TTopologyTemplate.java:123]: Why do do need to get a NT by name?

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/BackendUtils.java [hunk 19]
```diff
@@ -767,55 +767,39 @@ public static void persist(Object o, RepositoryFileReference ref, MediaType medi
             LSInput input = new LSInput() {
 
                 @Override
-                public void setSystemId(String systemId) {
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/BackendUtils.java:770]: Why do you have  changes here? It does not seem to be any real change...

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/BackendUtils.java [hunk 20]
```diff
@@ -1749,4 +1745,27 @@ public static void commit(DefinitionsChildId componentToCommit, String commitMes
...
+     * <p>
+     * Similar functionality as {@link IRepository#getAllDefinitionsChildIds(java.lang.Class)}, but it crawls through
+     * the repository
+     * <p>
+     * This method is required as we do not use a database.
+     *
+     * @param idClass class of the Ids to search for
+     * @return empty set if no ids are available
+     */
+    public <T extends ToscaElementId> SortedSet<T> getAllTOSCAElementIds(Class<T> idClass) {
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/BackendUtils.java:1760]: Why did you move this code here? revert this change.

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/converter/support/DockerUtils.java [hunk 21]
```diff
@@ -0,0 +1,97 @@
...
+import org.eclipse.winery.repository.backend.IRepository;
+
+import org.apache.commons.lang3.StringUtils;
+
+public class DockerUtils {
+
+    public static String buildDockerImage(ArtifactTemplateId artifactTemplateId, IRepository repository) throws IOException {
+
+        TArtifactTemplate newDockerArtifactTemplate = repository.getElement(artifactTemplateId);
+
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/converter/support/DockerUtils.java:34]: remove these empty lines here between the vars

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/converter/support/DockerUtils.java [hunk 22]
```diff
@@ -0,0 +1,97 @@
...
+                referencedFile = dockerImageName + ".tar.gz";
+
+                final String compressedFile = Utils.compressTarFile(new File(folder + "/" + dockerImageName + ".tar"));
+                if (compressedFile != null) {
+
+                    final File folderFile = new File(folder);
+
+                    // Image has been compressed, clear the folder.
+
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/converter/support/DockerUtils.java:63]: remove the empty lines here also

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/converter/support/ScriptUtils.java [hunk 23]
```diff
@@ -0,0 +1,261 @@
...
+
+        newFile += "#//Self-contained CSAR//" + line + '\n';
+        newFile += "csarRoot=$(find ~ -maxdepth 1 -path \"*.csar\");";
+        newFile += "\n";
+        newFile += "IFS=';' read -ra NAMES <<< \"$DAs\";";
+        newFile += "\n";
+        newFile += "for i in \"${NAMES[@]}\"; do";
+        newFile += "\n";
+        newFile += "	IFS=',' read -ra PATH <<< \"$i\"; ";
+        newFile += "\n";
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/converter/support/ScriptUtils.java:213]: remove the "\n"s at 211, 213, 215, 217, 221,223, 225, 227, 229, 231, 237

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/converter/support/ScriptUtils.java [hunk 24]
```diff
@@ -0,0 +1,261 @@
...
+            new File(new File(filename).getParent()).mkdirs();
+        }
+        new File(filename).delete();
+        final FileWriter bw = new FileWriter(filename);
+        bw.write(content);
+        bw.close();
+    }
+
+    static public String updateScriptFile(String line) {
+        String newFile = "";
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/converter/support/ScriptUtils.java:203]: Please us a StringBuilder instead of a String directly

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/converter/support/ScriptUtils.java [hunk 25]
```diff
@@ -0,0 +1,261 @@
...
+    static public void downloadAptDependencies(String packet) {
+        Process proc;
+
+        final Runtime rt = Runtime.getRuntime();
+
+        final String cmd =
+            "apt-get download $(apt-cache depends --recurse --no-recommends --no-suggests --no-conflicts --no-breaks --no-replaces --no-enhances --no-pre-depends "
+                + packet + " | grep \"^\\w\")";
+
+        System.out.println("Command: " + cmd);
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/converter/support/ScriptUtils.java:250]: Please use a logger.

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/SelfContainmentUtil.java [hunk 26]
```diff
@@ -0,0 +1,243 @@
...
+        TDeploymentArtifacts deploymentArtifacts = new TDeploymentArtifacts.Builder(deploymentArtifactTemplatesList).build();
+
+        TNodeTypeImplementation nodeTypeImpl = repository.getElement(newNodeTypeImplId);
+        nodeTypeImpl.setDeploymentArtifacts(deploymentArtifacts);
+
+        // FIXME: If we need to add new script artifact templates as new Impl Artifacts, I will need help here: 
+//        Collection<DefinitionsChildId> implementationArtifactIds = artifactTemplateIds.get("ImplementationArtifacts");
+//        TImplementationArtifacts tmp = new TImplementationArtifacts();
+//        
+//        List<TImplementationArtifacts.ImplementationArtifact> buildList = new ArrayList<>();
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/SelfContainmentUtil.java:95]: If you don't need this code, please remove or fix it

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/SelfContainmentUtil.java [hunk 27]
```diff
@@ -0,0 +1,243 @@
...
+                        continue;
+                    }
+                } else if ("ScriptArtifact".equals(element.getType().getLocalPart())) {
+                    DefinitionsChildId selfContainedArtifactTemplateId = doesSelfContainedArtifactTemplateExist(childid, repository);
+                    if (selfContainedArtifactTemplateId != null) {
+                        deploymentArtifactList.add(selfContainedArtifactTemplateId);
+                        artifactTemplates.put("DeploymentArtifacts", deploymentArtifactList);
+
+                        // FIXME: What do we do with updated script artifact? Is it an implemenetation artifact for the node template implementation?
+//                        ArtifactTemplateId existingScriptArtifactTemplate = findExistingScriptArtifactTemplate(childid, repository);
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/SelfContainmentUtil.java:179]: see above

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/ToscaExportUtil.java [hunk 28]
```diff
@@ -82,31 +83,28 @@
...
+        this.processTOSCA(repository, id, new CsarContentProperties(id.getQName().toString()), conf, false);
         CsarEntry csarEntry = this.referencesToPathInCSARMap.values().stream()
             .filter(entry -> entry instanceof XMLDefinitionsBasedCsarEntry)
             .findFirst()
             .orElseThrow(() -> new RepositoryCorruptException("Definition not found!"));
         csarEntry.writeToOutputStream(outputStream);
     }
 
-    public enum ExportProperties {
-        REPOSITORY_URI
```
Comment (lharzenetter) [inline org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/ToscaExportUtil.java:95]: Why did you move this code? please revert this change

**Review by lharzenetter — CHANGES_REQUESTED** [review]
There are a lot of code moves without any changes. Please only commit the changes required for the feature.

File: org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/BackendUtils.java (hunk 1 above)
Comment (barlasemre) [reply org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/BackendUtils.java:1057]: I'll check this in the weekend

File: .idea/codeStyles/Project.xml (hunk 17 above)
Comment (barlasemre) [reply .idea/codeStyles/Project.xml:33]: I am not able revert it, it automatically commits this :/
